Sources/Tracking/CustomerIO.swift

```diff
@@ -3,6 +3,8 @@ import Foundation
 public protocol CustomerIOInstance: AutoMockable {
     var siteId: String? { get }
 
+    var identifier: String? { get }
```

Comment:
I think an in-memory instance of `identifier` isn't very safe as instances of `CustomerIOInstance` can be destroyed and re-created during the lifecycle of an app. 

Instead, I would use [`ProfileStore`](https://github.com/customerio/customerio-ios/blob/4db3dc097e3cf127264ef2d0830499bd10966b4d/Sources/Tracking/Store/ProfileStore.swift)

Reply:
this was backed by the `profileStore` owned by the `identifyRepository` - I moved it instead to be backed by one on `CustomerIOImplementation` - in either case, my understanding was that Swift didn't cache these values, but called out to the underlying method when defined, but maybe I'm wrong?

Reply:
> my understanding was that Swift didn't cache these values, but called out to the underlying method when defined, but maybe I'm wrong?

I think what you're asking is "Looking at the `ProfileStore` get/set, does Swift call get/set once and cache the returned result?"

```swift
    public var identifier: String? {
        get {
            // ???? Does Swift cache this returned result? 
            keyValueStorage.string(.identifiedProfileId)
        }
        set {
            // ???? Does Swift cache `newValue`?
            keyValueStorage.setString(newValue, forKey: .identifiedProfileId)
        }
    }
```

Since we have explicitly defined a get/set function, Swift will always call those functions. It's like `var identified` is a function, not a class field. 

One way to do a cache is: `lazy var ...`

Reply:
Since the get/set functions are using the thread-safe keyValueStorage class, we can use non-singleton instances of `ProfileStore` all over the SDK and we will always get the latest value across all instances.